framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -358,10 +363,11 @@ bool ServiceReferenceBasePrivate::UngetService(
     } catch (...) {
       std::string message("ServiceFactory threw an exception");
       registration->bundle->coreCtx->listeners.SendFrameworkEvent(
-        FrameworkEvent(FrameworkEvent::Type::FRAMEWORK_WARNING,
+        FrameworkEvent(FrameworkEvent::Type::FRAMEWORK_ERROR,
                        MakeBundle(bundle->shared_from_this()),
                        message,
-                       std::current_exception()));
+                       std::make_exception_ptr(ServiceException(message, 
```

Comment:
Since an exception was thrown, we want to preserve the exception message and use that string as the ```message``` in the ```ServiceException``` object. Otherwise we are losing information which may be useful to users.